### DISCUSSION TITLE
can you add memory and cpu usage info in side pane plus in each docker container section

### DISCUSSION THREAD

**Discussion by Greatz08** [louislam/dockge#378] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

if possible please add cpu and ram usage section for each container in left side panel which can be hide too if needed and in docker container section too add these things to get better idea of how much resources are consumed by containers in realtime  

### ✔️ Solution

:-)

**Comment by s4msy** [comment]
would be nice to see them in the same style as the uptime-kuma response-graphs. Because: recognition factor, CI etc.

**Comment by BluDood** [comment]
This would be a great feature, seems like [`docker stats`](https://docs.docker.com/engine/reference/commandline/container_stats/) could be used maybe?

**Reply by C-Fu** [reply]
Not mine, got it from https://stackoverflow.com/questions/47331106/enhanced-docker-stats-command-with-total-amount-of-ram-and-cpu

stats.sh
```
#!/bin/bash

# This script is used to complete the output of the docker stats command.
# The docker stats command does not compute the total amount of resources (RAM or CPU)

# Get the total amount of RAM, assumes there are at least 1024*1024 KiB, therefore > 1 GiB
HOST_MEM_TOTAL=$(grep MemTotal /proc/meminfo | awk '{print $2/1024/1024}')

# Get the output of the docker stat command. Will be displayed at the end
# Without modifying the special variable IFS the ouput of the docker stats command won't have
# the new lines thus resulting in a failure when using awk to process each line
IFS=;
DOCKER_STATS_CMD=`docker stats --no-stream --format "table {{.MemPerc}}\t{{.CPUPerc}}\t{{.MemUsage}}\t{{.Name}}"`

SUM_RAM=`echo $DOCKER_STATS_CMD | tail -n +2 | sed "s/%//g" | awk '{s+=$1} END {print s}'`
SUM_CPU=`echo $DOCKER_STATS_CMD | tail -n +2 | sed "s/%//g" | awk '{s+=$2} END {print s}'`
SUM_RAM_QUANTITY=`LC_NUMERIC=C printf %.2f $(echo "$SUM_RAM*$HOST_MEM_TOTAL*0.01" | bc)`

# Output the result
echo $DOCKER_STATS_CMD
echo -e "${SUM_RAM}%\t\t\t${SUM_CPU}%\t\t${SUM_RAM_QUANTITY}GiB / ${HOST_MEM_TOTAL}GiB\tTOTAL"

```
The result on my pi
```
# ./stats.sh
MEM %     CPU %     MEM USAGE / LIMIT     NAME
0.67%     0.25%     50.92MiB / 7.458GiB   speedtest-tracker
0.15%     0.00%     11.12MiB / 7.458GiB   rdesktop
1.70%     0.49%     129.6MiB / 7.458GiB   dockge
1.72%     0.14%     131.1MiB / 7.458GiB   runtipi-reverse-proxy
3.74%     1.45%     285.5MiB / 7.458GiB   runtipi
0.75%     10.26%    57.5MiB / 7.458GiB    runtipi-db
0.24%     0.45%     18.41MiB / 7.458GiB   runtipi-redis
2.80%     0.17%     213.9MiB / 7.458GiB   bazarr
5.10%     0.04%     389.8MiB / 7.458GiB   budge
0.41%     0.00%     31.48MiB / 7.458GiB   netbox
1.99%     0.01%     151.8MiB / 7.458GiB   jackett
2.23%     1.12%     170.6MiB / 7.458GiB   radarr
1.05%     0.00%     79.9MiB / 7.458GiB    homepage
1.33%     0.25%     101.4MiB / 7.458GiB   changedetection
2.75%     0.03%     210.3MiB / 7.458GiB   overseerr
3.74%     4.41%     285.4MiB / 7.458GiB   kavita
30.37%                  19.07%          2.26GiB / 7.45768GiB    TOTAL
```



If I comment the 2nd last line

```
# ./totalstats.sh
30.39%                  12.78%          2.27GiB / 7.45768GiB    TOTAL
```

**Comment by C-Fu** [comment]
Just a quick 2 min MS Paint masterpiece 
It's not exactly important, but this can save me a few clicks and a window/tab change
![image](https://github.com/louislam/dockge/assets/6426597/d4053f72-48da-4728-9821-9a1a597c8b95)

**Comment by JinbaIttai** [comment]
That's a handy script, thanks!

Slight improvement: pipe the docker stats command through "sort" to more easily identify memory hogs.

ie:

```bash
DOCKER_STATS_CMD=`docker stats --no-stream --format "table {{.MemPerc}}\t{{.CPUPerc}}\t{{.MemUsage}}\t{{.Name}}"|sort -n`
```

**Comment by Nic-kh** [comment]
Maybe it will be useful for someone, I use `ctop` for this - very convenient

**Comment by justwiebe** [comment]
@louislam I'd love to work on this if it would be a feature you'd accept

**Reply by justwiebe** [reply]
I couldn't help myself and started on this feature. I have a draft PR here; hopefully this weekend I'll get time to go through the rest of the checklist https://github.com/louislam/dockge/pull/700

If you like it, I can make an index page to show all of the docker stats

**Comment by DoTheEvo** [comment]
+1
I would like to get more people in to managing docker hosts I setup, and a web interface would make it easier.. and to see that some container is using 100% cpu would be really helpful.
I am using [ctop](https://github.com/bcicen/ctop) when in terminal and I absolutely love it. 
And would love to love dockge if it would have those basic ctop features when it comes to getting the overview of how stuff is running on a docker host.
